app/frontend/src/components/Navigation.tsx

```diff
@@ -1,113 +1,162 @@
 import {
-  Collapse,
+  AppBar,
+  Button,
+  Grid,
   Hidden,
-  List,
-  ListItem,
-  ListItemText,
+  Toolbar,
+  Typography,
   makeStyles,
 } from "@material-ui/core";
-import React, { useState } from "react";
-import { Link } from "react-router-dom";
+import React from "react";
+import { Link, NavLink } from "react-router-dom";
 import {
   profileRoute,
   messagesRoute,
   requestsRoute,
   logoutRoute,
 } from "../AppRoutes";
 import { useTypedSelector } from "../store";
-import classNames from "classnames";
 import SearchBox from "./SearchBox";
 
+const menu = [
+  {
+    name: "Dashboard",
+    route: "/"
+  },
+  {
+    name: "Messages",
+    route: messagesRoute
+  },
+  {
+    name: "Requests",
+    route: requestsRoute
+  },
+  {
+    name: "Profile",
+    route: profileRoute
+  },
+];
+
 const useStyles = makeStyles((theme) => ({
-  listContainer: {
-    display: "block",
-    width: "100%",
-    [theme.breakpoints.up("md")]: {
-      display: "flex",
-      flexWrap: "wrap",
+  appBar: {
+    top: 0,
+    bottom: 'auto',
+    [theme.breakpoints.down("sm")]: {
+      bottom: 0,
+      top: 'auto',
     },
   },
-  menuList: {
+  search: {
+    flexGrow: 3,
+    display: 'flex',
+    justifyContent: 'flex-end',
+  },
+  title: {
+    fontWeight: "bold",
+  },
+  gutters: {
     [theme.breakpoints.down("sm")]: {
-      marginLeft: "1em",
+      paddingLeft: 0,
+      paddingRight: 0,
+    },
+  },
+  label: {
+    fontSize: '0.7rem',
+  },
+  flex: {
+    width: 'auto',
+    flex: 0,
+    padding: '0 3rem',
+    [theme.breakpoints.down("sm")]: {
```

Comment:
As we are developing mobile-first, it is better to define the mobile layout as the default. And specify a separate up for everything wider.